src/lib/modules/job/job.go

```diff
@@ -0,0 +1,189 @@
+package job
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	config "github.com/bench-routes/bench-routes/src/lib/config_v2"
+	"github.com/bench-routes/bench-routes/src/lib/modules/evaluate"
+	"github.com/bench-routes/bench-routes/tsdb/file"
+)
+
+// Executable is an interface that is implemented by machineJob and monitoringJob
+type Executable interface {
+	Execute()
+	Abort()
+	Info() *JobInfo
+}
+
+type JobInfo struct {
+	mux         sync.RWMutex
+	Name        string
+	Every       time.Duration
+	lastExecute time.Time
+}
+
+// NewJob creates a new job based on the typ
+func NewJob(typ string, app file.Appendable, c chan struct{}, api *config.API) (Executable, error) {
+	switch typ {
+	case "machine":
+		job, err := newMachineJob(app, c, api)
+		if err != nil {
+			return nil, fmt.Errorf("error creating job : %w", err)
+		}
+		return job, nil
+	case "monitor":
+		job, err := newMonitoringJob(app, c, api)
+		if err != nil {
+			return nil, fmt.Errorf("error creating job : %w", err)
+		}
+		return job, nil
+	default:
+		return nil, fmt.Errorf("`typ` provided is invalid")
+	}
+}
+
+type monitoringJob struct {
+	JobInfo
+	app     file.Appendable
+	sigCh   chan struct{}
+	client  *http.Client
+	request *http.Request
+}
+
+func newMonitoringJob(app file.Appendable, c chan struct{}, api *config.API) (*monitoringJob, error) {
```

Comment:
```suggestion
func newMonitoringJob(app file.Appendable, api *config.API) (*monitoringJob, chan <- struct{}, error) {
```
Lets create a channel here and pass it to the caller function. This will be a good abstraction.

Reply:
If we create a channel here and pass it down to low-level implementations then it would work right, but then higher level modules(scheduler package) won't be able to access and signal the channel. That would create an implementation issue for the scheduler.